lantiq: remove bogus PCI ID + NVMEM

neheb opened the pull request:
ping @hauke @robimarko @lmadam20 @xdarklight

neheb committed:
lantiq: fritz7430: fix PCI ID

This platform has a bogus PCI ID for the chip. Correct it in DTS.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16555
Signed-off-by: Robert Marko <robimarko@gmail.com>

neheb commented:
Another example of this is: https://github.com/openwrt/openwrt/blob/main/target/linux/ath79/dts/qca9563_netgear_wndr.dtsi#L274

I guess the compatible string really does override.

hauke commented on package/kernel/mac80211/patches/ath9k/513-ath9k_add_pci_ids.patch:
```diff
@@ -1,30 +0,0 @@
- 	case AR5416_DEVID_PCIE:
- 	case AR5416_AR9100_DEVID:
---- a/drivers/net/wireless/ath/ath9k/hw.h
-+++ b/drivers/net/wireless/ath/ath9k/hw.h
```
There was a discussion about this PCI ID in this patch already:
https://patchwork.ozlabs.org/project/openwrt/patch/20220116213025.1379508-1-hauke@hauke-m.de/

neheb replied:
That link is to a TD-W8970, which has

compatible = "pci168c,002e";

already.

Although the node is wrong: pci168c,002e  # AR9287 according to Documentation/devicetree/bindings/net/wireless/qca,ath9k.yaml

pci168c,0030 is the correct one.

neheb replied:
Added commit to fix this.

neheb committed:
lantiq: use nvmem for tplink tdw89x0

These two devices use different wifi chips and as a result, different
calibration sizes. Move the differences out of dtsi.

Signed-off-by: Rosen Penev <rosenp@gmail.com>
Link: https://github.com/openwrt/openwrt/pull/16555
Signed-off-by: Robert Marko <robimarko@gmail.com>

neheb commented:
Rebased.

Sort of depends on https://github.com/openwrt/openwrt/pull/9775

neheb commented:
@hauke removed the mac80211 patch. Looks like it needs to remain. Now this is only an NVMEM conversion and compatible line fix.

robimarko commented:
Thanks! Rebased on top of main and merged!
